currency/forexprovider/currencylayer/currencylayer_test.go

```diff
@@ -1,137 +1,129 @@
 package currencylayer
 
 import (
+	"log"
+	"os"
+	"strings"
 	"testing"
 
+	"github.com/stretchr/testify/assert"
 	"github.com/thrasher-corp/gocryptotrader/currency/forexprovider/base"
 )
 
 var c CurrencyLayer
 
-// please set your API key here for due diligence testing NOTE be aware you will
-// minimize your API calls using this test.
-const (
-	APIkey   = ""
-	Apilevel = 0
+// Either set your API key here or in env var for integration testing
+var (
+	apiKey      = ""
+	apiKeyLevel = 0
```

Comment:
So many functions are behind a check of 
```go
	if c.APIKeyLvl >= AccountBasic {
		return 0, errors.New("insufficient API privileges, upgrade to basic to use this function")
	}
```
It was very confusing, but for anyone else that tests, leave this as `0` 😄

Reply:
Yeah. I'd actually like to remove APIKeyLvl across the board, because I'm not seeing anywhere we'd base a decision tree off it that would be much better than letting the user hit the API's own error.
That is: Yes, we can pre-empt the error, but unless we're going to save requests that the engine or syncer would make wastefully, then there more downside in trying to clientside their pricing plan.
As I've seen it so far, every feature locked in API levels beyond basic is not a feature we'd use *automatically*, and the user can manage their own API calls.
Anyway. Out of scope and not tackling.